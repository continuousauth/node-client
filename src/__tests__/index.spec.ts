import { getOtp } from '../index';
import { requestThroughCircleCI } from '../circleci';

jest.mock('../circleci');

const mockedRequestThroughCircleCI = requestThroughCircleCI as jest.Mock;

describe('@continuous-auth/client', () => {
  let env: NodeJS.ProcessEnv;

  beforeEach(() => {
    env = { ...process.env };
  });
  afterEach(() => {
    process.env = env;
  });

  it('should throw an error when CFA_SECRET is not set', async () => {
    await expect(getOtp()).rejects.toMatchSnapshot();
  });

  it('should throw an error when CFA_PROJECT_ID is not set', async () => {
    process.env.CFA_SECRET = 'test';
    await expect(getOtp()).rejects.toMatchSnapshot();
  });

  it('should throw an error on an unsupported CI platform', async () => {
    process.env.CFA_SECRET = 'test';
    process.env.CFA_PROJECT_ID = '123';
    delete process.env.GITHUB_ACTIONS;
    await expect(getOtp()).rejects.toMatchSnapshot();
  });

  it('should use the circleci provider on circleci', async () => {
    process.env.CFA_SECRET = 'secret';
    process.env.CFA_PROJECT_ID = '123';
    process.env.CIRCLECI = 'true';
    mockedRequestThroughCircleCI.mockReturnValue(Promise.resolve({ response: 'example token' }));
    const token = await getOtp();
    expect(token).toEqual('example token');
  });
});
