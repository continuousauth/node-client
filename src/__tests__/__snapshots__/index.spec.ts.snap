// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@continuous-auth/client should throw an error on an unsupported CI platform 1`] = `[Error: Unsupported CI provider, currently we only support CircleCI and GitHub Actions]`;

exports[`@continuous-auth/client should throw an error when CFA_PROJECT_ID is not set 1`] = `[Error: Requested env var "CFA_PROJECT_ID" is missing or empty]`;

exports[`@continuous-auth/client should throw an error when CFA_SECRET is not set 1`] = `[Error: Required env var "CFA_SECRET" is missing or empty]`;
